src/twisted/mail/test/test_mailmail.py

```diff
@@ -73,3 +102,252 @@ def test_copyAliasedSender(self):
         """
         exc = self.assertRaises(SystemExit, parseOptions, ['-om'])
         self.assertEqual(exc.args, ("Unsupported option.",))
+
+
+    def test_version(self):
+        """
+        The I{--version} option displays the version and raises
+        L{SystemExit}.
+        """
+        out = NativeStringIO()
+        self.patch(sys, 'stdout', out)
+        systemExitCode = self.assertRaises(SystemExit, parseOptions,
+                                           '--version')
+        # SystemExit.code is None on success
+        self.assertEqual(systemExitCode.code, None)
+        data = out.getvalue()
+        self.assertEqual(data, "mailmail version: {}\n".format(version))
+
+
+    def test_backGroundDelivery(self):
+        """
+        The I{-odb} flag specifies background delivery.
+        """
+        stdin = NativeStringIO('\n')
+        self.patch(sys, 'stdin', stdin)
```

Comment:
do we need to patch stdin? I don't see any check on the patched stdin and I don't see how it affects the test

re-reading the code I see why this is needed, but maybe is better to refactor these 2 lines into something like

self.sendNoEmail() or self.prepareEmptySend or something like that.

Reply:
we need to patch stdin, because of this code in mailmail:

```
o.body = NativeStringIO(buffer.getvalue() + sys.stdin.read())
```
If you don't patch stdin, then the test just hangs, because it is waiting for stdin input.